tests/test_api.py

```diff
@@ -467,6 +470,158 @@ def test_create_specification_auth(testclient):
+        response.raise_for_status()
+        r = schema.APIGetBuild.parse_obj(response.json())
+        assert r.status == schema.APIStatus.OK
+        assert r.data.specification.name == environment_name
+
+        # Exits immediately on failure
+        assert r.data.status != 'FAILED'
+
+        # If not done, adds the id back to the end of the queue
+        if r.data.status != 'COMPLETED':
+            build_ids.append(build_id)
+
+        # Adds a small delay to avoid making too many requests too fast. The
+        # build takes significantly longer than 1 second, so this shouldn't
+        # impact the measurements
+        time.sleep(1)
+
+    # If there are jobs in the queue, the loop didn't complete in the allocated
+    # time. So something went wrong, like a build getting stuck or parallel
+    # builds not working
+    assert len(build_ids) == 0
+
+    # Because this is an integration test, we cannot change the server
+    # c.CondaStoreWorker.concurrency value, which is set to 4 by default. But
+    # it's possible to devise a statistical test based on locally collected
+    # data, using 'build_deltas' above:
+    #
+    # concurrency = 4 with 2 CPUs, so equivalent to concurrency = 2:
+    # c4_2cpu = [
+    #     1.027987,
+    #     67.234371,
+    #     1.272288,
+    #     43.966526,
+    #     7.171627,
+    #     68.563222,
+    #     4.143675,
+    #     46.872263,
+    #     1.018258,
+    # ]
+    #
+    # concurrency = 1, same machine:
+    # c1 = [
+    #     19.394085,
+    #     33.70623,
+    #     22.815429,
+    #     62.555845,
+    #     68.438333,
+    #     29.794979,
+    #     63.370743,
+    #     32.118421,
+    #     29.88376,
+    # ]
+    #
+    # Here's another set of measurements from a different machine, which should
+    # have 4 CPUs, with concurrency = 4:
+    # ci = [
+    #     1.02644,
+    #     33.591736,
+    #     1.016269,
+    #     19.299738,
+    #     7.115025,
+    #     20.342442,
+    #     12.222805,
+    #     6.103751,
+    #     1.016237,
+    # ]
+    #
+    # These values will vary depending on workload and the number of CPUs. But
+    # the main observation here is this: if parallel builds are working,
+    # there will be a number of values that are relatively small compared to the
+    # time it takes to run a single build when not running concurrently.
+    #
+    # So the test below looks at the value of the first quartile, which is the
+    # median of the lower half of the dataset, where all these small values will
+    # be located, and compares it to a certain threshold, which is unlikely to
+    # be reached based on how long it takes a single build to run
+    # non-concurrently on average:
+    threshold = 5
+    quartiles = statistics.quantiles(build_deltas, method='inclusive')
+    print("build_deltas", build_deltas)
+    print("stats", min(build_deltas), quartiles)
+    assert quartiles[0] < threshold
```

Comment:
I ran benchmarks on a completely different machine with 4 CPUs, just to see how values would change. I think the above test is okay based on these:

concurrency = 4:
```
build_deltas [1.025565, 7.176997, 1.019974, 77.987058, 4.085229, 25.557227, 14.347968, 8.206779, 2.057178]
stats 1.019974 [2.057178, 7.176997, 14.347968]

build_deltas [1.031824, 1.020348, 45.126866, 37.897655, 17.460182, 8.242122, 65.296544, 1.02701, 1.049515]
stats 1.020348 [1.031824, 8.242122, 37.897655]

build_deltas [1.029401, 50.235828, 1.067922, 31.738958, 17.412726, 41.970197, 28.651753, 1.030208, 14.338367]
stats 1.029401 [1.067922, 17.412726, 31.738958]
```
(The last one has a bigger median because the retrieve packages job was running in parallel.
I wanted to see how the numbers would be affected by that. Note that the median of the lower half was not affected.)

----

concurrency = 1:
```
build_deltas [28.631925, 33.738711, 29.625621, 56.210018, 57.21373, 33.693227, 53.14048, 33.745332, 35.830624]
stats 28.631925 [33.693227, 33.745332, 53.14048]

>       assert quartiles[0] < threshold
E       assert 33.693227 < 5

build_deltas [28.665233, 33.790378, 29.662559, 56.242397, 57.218049, 33.758011, 105.30695, 33.760865, 35.784077]
stats 28.665233 [33.758011, 33.790378, 56.242397]

>       assert quartiles[0] < threshold
E       assert 33.758011 < 5

build_deltas [37.866588, 33.755057, 29.649771, 50.167768, 67.47894, 33.751589, 56.277974, 37.855099, 36.887129]
stats 29.649771 [33.755057, 37.855099, 50.167768]

>       assert quartiles[0] < threshold
E       assert 33.755057 < 5
```

Reply:
And here's a [box plot](https://en.wikipedia.org/wiki/Box_plot#Elements) of the datasets from the test comment. Note how the Q1 value (the lower part of the box that represents the median of the smallest half of the dataset) is very similar between `c4_2cpu` and `ci` (which are running with concurrent builds on). Q1 is very close to the min value and the min is itself very small. Compare that to `c1`, which is what we're trying to detect, which is running non-concurrently.

![tests](https://github.com/conda-incubator/conda-store/assets/10469885/28ffa79d-902a-41b1-addc-2205ed6adce8)

```py
import matplotlib
matplotlib.use('TkAgg')
import matplotlib.pyplot as plt

my_dict = {'c4_2cpu': c4_2cpu, 'c1': c1, 'ci': ci}
plt.boxplot(my_dict.values(), labels=my_dict.keys())
plt.show()
```

Reply:
And based on the other benchmark values posted above, which are from a different machine, you can see that the smallest half doesn't vary much when running concurrently, so it should be reliable as an estimate.

Reply:
The reason Q1 is used (as opposed to just comparing min values) is because it makes use of the smallest half of the dataset, which is what we're interested in, and the median value helps filter out outliers there.